server-common/src/main/java/org/apache/kafka/deferred/DeferredEventQueue.java

```diff
@@ -77,13 +76,13 @@ void failAll(Exception exception) {
      * @param offset        The offset to add the new event at.
      * @param event         The new event.
      */
-    void add(long offset, DeferredEvent event) {
+    public void add(long offset, DeferredEvent event) {
         if (!pending.isEmpty()) {
             long lastKey = pending.lastKey();
             if (offset < lastKey) {
                 throw new RuntimeException("There is already a purgatory event with " +
```

Comment:
How about throwing an `IllegalArgumentException` instead of `RuntimeException` with the following message:
> There is already a deferred event with offset ...